### PR TITLE
media-tv/kodi: Add "system-ffmpeg" use flag

### PR DESCRIPTION
**Pull request by candrews**
https://bugs.gentoo.org/show_bug.cgi?id=608472

### CODE REVIEW

**Comment by aballier** [comment]
I think this is a terrible idea: ffmpeg has regular security fixes which happen much more often than kodi releases. Building an almost identical copy for kodi seems also a waste of ressources.

Is there any single advantage in using bundled ffmpeg ?

File: media-tv/kodi/kodi-17.0-r1.ebuild [hunk 1]
```diff
@@ -214,7 +216,7 @@ src_configure() {
 		-DENABLE_DBUS=$(usex dbus)
 		-DENABLE_DVDCSS=$(usex css)
 		-DENABLE_INTERNAL_CROSSGUID=OFF
-		-DENABLE_INTERNAL_FFMPEG=OFF
+		-DENABLE_INTERNAL_FFMPEG="$(usex system-ffmpeg OFF ON)"
```
Comment (candrews) [inline media-tv/kodi/kodi-17.0-r1.ebuild]: so obvious :)

**Comment by candrews** [comment]
> Is there any single advantage in using bundled ffmpeg ?

By default, the system-ffmpeg USE flag is enabled. So unless you explicitly specify otherwise, system ffmpeg will continue to be used.

There are a few other packages in Gentoo right now that have a system-ffmpeg use flag. It doesn't really hurt to have the option (again, it's important as you said to use system libraries unless explicit instructions otherwise are specified) and it can be helpful if you're reporting issues to Kodi upstream (they really like you to use their bundled libraries when reporting issues to them) or you want to have installed a version of ffmpeg that doesn't work with Kodi (as happened with the ffmpeg 3 transition).

**Comment by aballier** [comment]
> There are a few other packages in Gentoo right now that have a system-ffmpeg use flag.

Those packages violate the policy of not using bundled libraries :)
Those are also exceptions, and as such, have very convincing reasons to do so and their maintainers are supposed to handle and track ffmpeg security issues by themselves.

> It doesn't really hurt to have the option (again, it's important as you said to use system libraries unless explicit instructions otherwise are specified)

It does hurt if the tons of bugfixes and security fixes in ffmpeg are not proactively backported.

> and it can be helpful if you're reporting issues to Kodi upstream (they really like you to use their bundled libraries when reporting issues to them)

Yeah, I know that :) But they also worked hard to unbundle it properly. For reporting issues that you encounter on Gentoo, you should go to Gentoo's bugzilla. If you want to skip that part, then better clone their repository and build manually with their recommended build options to avoid any other Gentoo patch or mandatory option. I don't see how bundling helps here.


> or you want to have installed a version of ffmpeg that doesn't work with Kodi (as happened with the ffmpeg 3 transition).

Those are handled by leaving breaking ffmpeg versions under package.mask until packages are fixed.

**Comment by SoapGentoo** [comment]
@candrews could you rebase on master?

**Comment by candrews** [comment]
@SoapGentoo

**Comment by SoapGentoo** [comment]
@aballier so what's the verdict? package.use.force `system-ffmpeg`? then people with real dedication can still disable it, but they need to jump through hoops, which seems like a middle ground here.

**Comment by aballier** [comment]
I have not seen any good reason to have the flag yet, hence I'd say it's not worth the effort at all.
p.u.force is the very least that can be done.

Also:
```
$ cat xbmc-17.1rc1-Krypton/tools/depends/target/ffmpeg/FFMPEG-VERSION 
LIBNAME=ffmpeg
BASE_URL=https://github.com/xbmc/FFmpeg/archive
VERSION=3.1.6-Krypton
ARCHIVE=$(LIBNAME)-$(VERSION).tar.gz
GNUTLS_VER=3.4.14
```

And from http://ffmpeg.org/security.html : 
```
3.1.7
Fixes following vulnerabilities:

CVE-2017-5024, 02a5e88ebc725b09f675bfcbbd4db1133e41708e / 2d453188c2303da641dafb048dc1806790526dfd
CVE-2017-5025, b6efd022b77349f2797afe756b791e82ec4a1d96 / fd30e4d57fe5841385f845440688505b88c0f4a
```

Is xbmc possibly affected by those ? Why ?
(reiterate this analysis every time a sec. vuln. is reported on ffmpeg)

**Pull request closed without merging**